tap_linkedin_ads/sync.py

```diff
@@ -1,444 +1,89 @@
-                                    parent_id,
-                                    child_total_records)
-
-        # Pagination: Get next_url
-        next_url = None
-        links = data.get('paging', {}).get('links', [])
-        for link in links:
-            rel = link.get('rel')
-            if rel == 'next':
-                href = link.get('href')
-                if href:
-                    next_url = 'https://api.linkedin.com{}'.format(urllib.parse.unquote(href))
-
-        LOGGER.info('%s: Synced page %s, this page: %s. Total records processed: %s',
-                    stream_name,
-                    page,
-                    record_count,
-                    total_records)
-        page = page + 1
-
-    # Write child bookmarks
-    for key, val in list(child_max_bookmarks.items()):
-        write_bookmark(state, key, val)
-
-    return total_records, max_bookmark_value
-
-
-# Review catalog and make a list of selected streams
-def get_selected_streams(catalog):
-    selected_streams = set()
-    for stream in catalog.streams:
-        mdata = metadata.to_map(stream.metadata)
-        root_metadata = mdata.get(())
-        if root_metadata and root_metadata.get('selected') is True:
-            selected_streams.add(stream.tap_stream_id)
-    return list(selected_streams)
-
-
-# Currently syncing sets the stream currently being delivered in the state.
-# If the integration is interrupted, this state property is used to identify
-#  the starting point to continue from.
-# Reference: https://github.com/singer-io/singer-python/blob/master/singer/bookmarks.py#L41-L46
 def update_currently_syncing(state, stream_name):
+    """
+    Currently syncing sets the stream currently being delivered in the state.
+    If the integration is interrupted, this state property is used to identify the
+    starting point to continue from.
+
+    Reference: https://github.com/singer-io/singer-python/blob/master/singer/bookmarks.py#L41-L46
+    """
     if (stream_name is None) and ('currently_syncing' in state):
+        # Remove the existing currently_syncing stream from the state for the complete sync
         del state['currently_syncing']
     else:
+        # Set currently_syncing stream
         singer.set_currently_syncing(state, stream_name)
     singer.write_state(state)
 
+def get_streams_to_sync(selected_streams):
+    """
+    Get lists of streams to call the sync method.
+    For children, ensure that dependent parent_stream is included even if it is not selected.
+    """
+    parent_streams = []
+
+    # Loop thru all selected streams
+    for stream_name in selected_streams:
+        stream_obj = STREAMS[stream_name]
+        # If the stream has a parent_stream, then it is a child stream
+        parent_stream = hasattr(stream_obj, 'parent') and stream_obj.parent
+
+        # Append selected parent streams
+        if not parent_stream:
+            parent_streams.append(stream_name)
+        elif parent_stream:
```

Comment:
Can we make it simple `if-else` instead of `if-elif`?

Reply:
Replaced `if-elif` with `if-else`.